common/persistence/persistenceInterface.go

```diff
@@ -368,19 +368,28 @@ type (
 
 	// InternalResetMutableStateRequest is used to reset workflow execution state  for Persistence Interface
 	InternalResetMutableStateRequest struct {
-		PrevRunID        string
+		// previous workflow information
+		PrevRunID            string
```

Comment:
Does it make sense to create a separate persistence type for a "WorkflowRun" - that contains (runID, lastWriteVersion, state) ? Looks like these params are needed / passed everywhere for assertion.

Reply:
created an issue for this: #2013